pkg/client/unversioned/clientcmd/merged_client_builder.go

```diff
@@ -112,8 +112,11 @@ func (config *DeferredLoadingClientConfig) ClientConfig() (*restclient.Config, e
 		//   "empty due to defaults"
 		// TODO: this shouldn't be a global - the client config rules should be
 		//   handling this.
-		defaultConfig, err := DefaultClientConfig.ClientConfig()
-		if err == nil && !reflect.DeepEqual(mergedConfig, defaultConfig) {
+		defaultConfig, defErr := DefaultClientConfig.ClientConfig()
+		if IsConfigurationInvalid(defErr) && !IsEmptyConfig(err) {
```

Comment:
> A configuration that is both default and non-empty must still be checked. DefaultClientConfigs are only sometimes empty.

This code needs a major overhaul. I think it doesn't handle a few cases. But the question is, should that block the release?

> So you still need to do the check against the default config in the non empty (err == nil) case. 

This is already being done by `DeepEqual` below.

> IsConfigurationInvalid covers IsEmptyConfig already.

No, they don't? `IsConfigurationInvalid` check is for the `DefaultConfig`. `IsEmptyConfig` is for the original config, i.e. `mergedConfig`. They are both calling different error objects, `defErr` and `err` respectively.